From: Matthias Brugger <mbrugger@suse.com>
Date: Tue, 20 Mar 2018 13:56:21 +0000
Subject: arm64: fix smccc compilation
Patch-mainline: Never, fixes local compilation issue
References: bsc#1068032

ARM_SMCC_CALL_VAL is a assembly macro. Fix compilation of
arm-smccc.h by setting ifdefs as in mainline.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>---
 arch/arm64/kernel/smccc-call.S |    1 +
 include/linux/arm-smccc.h      |    9 ++++++---
 2 files changed, 7 insertions(+), 3 deletions(-)

--- a/arch/arm64/kernel/smccc-call.S
+++ b/arch/arm64/kernel/smccc-call.S
@@ -12,6 +12,7 @@
  *
  */
 #include <linux/linkage.h>
+#include <linux/arm-smccc.h>
 #include <asm/asm-offsets.h>
 
 	.macro SMCCC instr
--- a/include/linux/arm-smccc.h
+++ b/include/linux/arm-smccc.h
@@ -14,9 +14,6 @@
 #ifndef __LINUX_ARM_SMCCC_H
 #define __LINUX_ARM_SMCCC_H
 
-#include <linux/linkage.h>
-#include <linux/types.h>
-
 #include <uapi/linux/const.h>
 
 /*
@@ -80,6 +77,11 @@
 			   ARM_SMCCC_SMC_32,				\
 			   0, 0x8000)
 
+#ifndef __ASSEMBLY__
+
+#include <linux/linkage.h>
+#include <linux/types.h>
+
 /**
  * struct arm_smccc_res - Result from SMC/HVC call
  * @a0-a3 result values from registers 0 to 3
@@ -262,4 +264,5 @@ asmlinkage void arm_smccc_hvc(unsigned l
  */
 #define arm_smccc_1_1_hvc(...)	__arm_smccc_1_1(SMCCC_HVC_INST, __VA_ARGS__)
 
+#endif /*__ASSEMBLY__*/
 #endif /*__LINUX_ARM_SMCCC_H*/
